test/controllers/SetupDataControllerSpec.scala

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import base.BaseSpec
+import mocks.MockDataRepository
+import models.DynamicDataModel
+import play.api.libs.json.Json
+import play.api.test.FakeRequest
+import play.mvc.Http.Status
+
+class SetupDataControllerSpec extends BaseSpec with MockDataRepository {
+
+  object TestSetupDataController extends SetupDataController(mockDataRepository)
+
+  "SetupDataController.addData" when {
+
+    "validateUrlMatch returns 'true'" should {
```

Comment:
There is no function called validUrlMatch but this test case makes it look like there is. Could you reword to say what is happening instead please?